### ISSUE TITLE
Optional language field for articles

### ISSUE THREAD

**Issue by abenerd** [forem/forem#4168]
**Is your feature request related to a problem? Please describe.**
Nope, no problem!

**Describe the solution you'd like**
An optional language field for an article so one can search or filter articles by their language.

**Describe alternatives you've considered**
One could use the normal existing tags for that, but I think an actual designated tag on the article might be better. 

**Additional context**
None that I can think of right now!

**Comment by EngineerCoding** [comment]
I really look forward to have something like this implemented, as I am actually slightly irritated by articles which or in other languages other than English (even if they are in my native language, Dutch). I even would like to propose that there should be a setting in your profile which sets in which languages you would like to receive in your feed. This should not be limited to a single a language, and when left blank or unspecified no additional filtering should be applied.

Once you are searching for something however, there should be an option to (temporarily) ignore this setting or default to your profile settings so that you can still find articles in other languages.

So, in a nutshell, I would like to extend this issue with the following:

- Be able to select the languages (note the plural here) you would like to receive in your feed. If left unspecified, no filtering should be applied
- When searching, the language filter defaults to your profile settings, but can be changed to find articles in other languages

I would love to get started on this myself, but unfortunately I don't have a lot of time at the moment and furthermore I don't really know Ruby to be quite honest. At the very least I hope to contribute this way to get the discussion started for a proper implementation.

**Comment by michael-tharrington** [comment]
I think this is a great idea! 

To sum up a recent convo with a fellow DEV member, they mentioned that the tag system is imperfect for language for a number of reasons, but one not-so-obvious one is that users can only use 4 tags on their post, and if they have to designate their posts with a language (e.g. "espanol") that's one less tag they can use for topical reach.

**Comment by CodeMouse92** [comment]
I'd like to add my vote to this. We have excellent posts like [this one](https://dev.to/arielbarcellos/seja-mais-produtivo-com-linux-1-2c18) which are not in English, but come through English-only feeds like mine. As a moderator, I have no way to move these out of "English" into another language, so having something *besides* a tag to indicate post language, and possibly allowing moderators to modify this, would be very helpful.

**Comment by badiali** [comment]
#9581
